Tests/Test.cs

```diff
@@ -0,0 +1,327 @@
+using System.Collections.Immutable;
+using FluentAssertions;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Time.Testing;
+
+namespace Tests;
+
+using static ShoppingCartCommand;
+using static ShoppingCartEvent;
+using static ShoppingCart;
+
+// Types Aliases
+using ProductItems = ImmutableDictionary<string, int>;
+using ProductItem = (Guid ProductId, int Quantity);
+using PricedProductItem = (Guid ProductId, int Quantity, decimal Price);
+
+// Commands
+public abstract record ShoppingCartCommand
+{
+    public record AddProduct(
+        ProductItem ProductItem,
+        Guid? ClientId = null
+    ): ShoppingCartCommand;
+
+    public record RemoveProduct(
```

Comment:
can I not remove an item if the price changes in between?
this doesnt make sense and is very distracting when someone is tryiing to see the bigger picture

Reply:
Price won't change here as it's frozen in the event.

Reply:
nosense! Prices change. Nobody locks anything just because someone put an electron into a cart